controller/codebase_blackbox_test.go

```diff
@@ -97,6 +96,50 @@ func (s *TestCodebaseREST) TestSuccessCreateCodebaseWithoutStackID() {
 	})
 }
 
+func (s *TestCodebaseREST) TestDeleteCodebase() {
+	resetFn := s.DisableGormCallbacks()
+	defer resetFn()
+
+	s.T().Run("OK", func(t *testing.T) {
+		resource.Require(t, resource.Database)
+
+		// Create space and codebase
+		cb := minimalCodebaseWithSpace(t, s.db, testsupport.TestIdentity.ID)
+
+		svc, ctrl := s.SecuredControllers(testsupport.TestIdentity)
+		test.DeleteCodebaseOK(t, svc.Context, svc, ctrl, cb.ID)
+	})
+
+	s.T().Run("NotFound", func(t *testing.T) {
+		resource.Require(t, resource.Database)
+
+		codebaseID := uuid.FromStringOrNil("d7a282f6-1c10-459e-bb44-55a1a6d48bdd")
+
+		svc, ctrl := s.SecuredControllers(testsupport.TestIdentity)
+		test.DeleteCodebaseNotFound(t, svc.Context, svc, ctrl, codebaseID)
+	})
+
+	s.T().Run("Unauthorized", func(t *testing.T) {
+		resource.Require(t, resource.Database)
+
+		codebaseID := uuid.FromStringOrNil("d7a282f6-1c10-459e-bb44-55a1a6d48bdd")
+
+		svc, ctrl := s.UnsecuredController()
+		test.DeleteCodebaseUnauthorized(t, svc.Context, svc, ctrl, codebaseID)
```

Comment:
@aslakknutsen I guess you meant difference between 401 and 404. In case of 401, we don't provide Identity and create unsecure controller while in case of 404 we provide identity and create secure controller.

Re: multiple tests

- Can I delete an existing codebase without auth? 401
May not required this test as we first check for auth irrelevant if resource exist or not. Check controller.Codebase.Delete(line no 124).  Do you see point to add it?

- Can I delete an existing codebase I don't own? 403
Already added, check "Forbidden" test, we are passing different Identity [testsupport.TestIdentity**2**] in delete call.

- Can I delete a non-existing codebase without auth? 401
Already added, check "Unauthorized" test.

Reply:
> May not required this test as we first check for auth irrelevant if resource exist or not. Check controller.Codebase.Delete(line no 124). Do you see point to add it?

The test should technically describe the API and intended behavior and not rely on impl details